src/modules/core/components/Fields/Input/InputComponent.tsx

```diff
@@ -37,7 +37,7 @@ type CleaveHTMLInputElement = HTMLInputElement & { rawValue: string };
 
 interface MaxButtonParams {
   setFieldValue: (field, value) => void;
-  maxAmount: string;
+  maxAmount: string | number;
```

Comment:
This is really confusing. Either one or the other.

As far as I understand you made this change because the type wouldn't work in your dialog form. But since this is a core component, the actual display component should adapt the types to it, not the other way around.

So please refactor this to have a more strict type.